cinema/views.py

```diff
@@ -48,6 +77,28 @@ class MovieSessionViewSet(viewsets.ModelViewSet):
     queryset = MovieSession.objects.all()
     serializer_class = MovieSessionSerializer
 
+    def get_queryset(self):
+        queryset = self.queryset
+        date = self.request.query_params.get("date")
+        movie = self.request.query_params.get("movie")
+
+        if date:
+            queryset = queryset.filter(show_time__date=date)
+
+        if movie:
+            queryset = queryset.filter(movie__id=movie)
+
+        if self.action == "list":
+            queryset = (
+                queryset
+                .select_related("movie", "cinema_hall")
+                .annotate(
+                    tickets_available=F("cinema_hall__rows")
+                    * F("cinema_hall__seats_in_row") - Count("tickets")
+                )
```

Comment:
Today we learn very nice method for Serializer
```
    tickets_available = serializers.SerializerMethodField(method_name="get_tickets_available")

    def get_tickets_available(self, obj):
        return obj.cinema_hall.capacity - len(obj.tickets.all())
```
I think it is more easy, but your choice